Eval.hs

```diff
@@ -444,6 +452,22 @@ isIdPair :: Val -> Bool
 isIdPair VIdPair{} = True
 isIdPair _         = False
 
+idComp :: Val -> Val -> Val -> Val -> Val -> Val -> Val
+idComp a u v w pId qId = case (pId, qId) of
+  (VIdPair p ps, qId) | eps `member` ps -> qId
+  (pId, VIdPair q qs) | eps `member` qs -> pId
```

Comment:
Hmmm, I'm not sure. I think that they are needed to make sure that `refl . p` will reduce even when `p` is just a variable. To hit the other case you need to know that `p` is in fact a `VIdPair`. The behaviour should agree when the cases overlap so I've now moved these two cases after the third one. But maybe this behaviour is too strong? Do we want `refl . p` to reduce in the case when `p` is an undetermined variables of type `Id A a b`?